Jint/Engine.cs

```diff
@@ -370,6 +383,17 @@ public Completion ExecuteStatement(Statement statement)
                 throw new TimeoutException();
             }
 
+#if NETCOREAPP2_0
+            if (Options._MemoryLimit > 0)
+            {
+                var memoryUsage = GC.GetAllocatedBytesForCurrentThread() - _initialMemoryUsage;
+                if (memoryUsage > Options._MemoryLimit)
+                {
+                    throw new MemoryLimitExceededException();
```

Comment:
I would suggest adding some details here. Script has allocated X but is limited to Y